apps/verticalhome/js/app.js

```diff
@@ -150,6 +164,206 @@
 
     },
 
+    saveAllPinnedApps: function(){
+      if(this.deletedApps.length>0){
+
+          for(var x = 0; x < app.pinnedAppsManager.items.length; x++){
+            if(app.pinnedAppsManager.items[x].element.style.display == 'none'){
+              app.pinnedAppsManager.items.splice(x,1);
+            }
+          }
+
+          for(var z = 0; z < this.deletedApps.length; z++){
+            app.itemStore.deletePinAppItem(this.deletedApps[z]);
+          }
+        }
+
+        for(var i = 0; i<app.pinnedAppsManager.items.length; i++){
+          var obj = {};
+
+          if(app.pinnedAppsManager.items[i].entry!=null){
+
+              obj.name = app.pinnedAppsManager.items[i].entry.name;
+              obj.manifestURL = typeof(
+                app.pinnedAppsManager.items[i].entry.manifestURL
+                ) == 'undefined'?
+                app.pinnedAppsManager.items[i].entry.targetApp.manifestURL
+                : app.pinnedAppsManager.items[i].entry.manifestURL;
+
+              obj.icon = typeof(
+                app.pinnedAppsManager.items[i].entry.icon
+                )=='undefined'?
+                app.pinnedAppsManager.items[i].entry.icon.baseURI
+                :app.pinnedAppsManager.items[i].entry.icon;
+
+              obj.index = app.pinnedAppsManager.items[i].index;
+
+              app.itemStore.savePinnedAppItem(obj);
+
+          }
+        }
+
+        var notif = document.getElementById('blinknotif');
+        notif.style.display = 'block';
+        notif.innerHTML = 'Changes saved';
+
+        setTimeout(function() {
+          notif.style.display = 'none';
+        }, 3000);
+
+    },
+
+    pin2MainScreen: function(){
+      var selectedMoreAppItem = document
+        .querySelector('#more-apps-screen .selected');
+
+      var selectedMoreAppItem_icon = selectedMoreAppItem
+        .getAttribute('data-test-icon');
+
+      var selectedMoreAppItem_data_identifier = selectedMoreAppItem
+        .getAttribute('data-identifier');
```

Comment:
It would be better to get 'data' attributes in another way.

selectedMoreAppItem.dataset.identifier
selectedMoreAppitem.dataset.testIcon